docs/jax.scipy.rst

```diff
@@ -29,6 +29,7 @@ jax.scipy.linalg
    eigh_tridiagonal
    expm
    expm_frechet
+   hilbert
```

Comment:
Just to be sure: I added this manually. If there is some mechanism that makes this change automatically, which I have missed, please let me know :)